components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/AbstractAPIManager.java

```diff
@@ -453,22 +453,16 @@ protected API getApi(GovernanceArtifact artifact) throws APIManagementException
 
     public API getAPI(APIIdentifier identifier) throws APIManagementException {
         try {
-            String organizationId = ApiMgtDAO.getInstance().getOrganizationIDByAPIUUID(identifier.getUUID());
+            String uuid;
+            if (identifier.getUUID() != null) {
+                uuid = identifier.getUUID();
+            } else {
+                uuid = apiMgtDAO.getUUIDFromIdentifier(identifier);
+            }
+            String organizationId = apiMgtDAO.getOrganizationIDByAPIUUID(uuid);
             Organization org = new Organization(organizationId);
             PublisherAPI publisherAPI = apiPersistenceInstance.getPublisherAPI(org, identifier.getUUID());
             API api = APIMapper.INSTANCE.toApi(publisherAPI);
-            if (api.isAsync()) {
-                api.setAsyncApiDefinition(getAsyncAPIDefinition(identifier));
-            } else {
-                api.setSwaggerDefinition(getOpenAPIDefinition(identifier, organizationId));
-            }
-            if (api.getType() != null && APIConstants.APITransportType.GRAPHQL.toString().equals(api.getType())){
-                api.setGraphQLSchema(getGraphqlSchema(api.getId()));
-            }
-            //check for API visibility
-            if (APIConstants.API_GLOBAL_VISIBILITY.equals(api.getVisibility())) { //global api
-                return api;
-            }
```

Comment:
is it needed to remove those?

Reply:
When checking the usages of swagger definition from this returned API there are none. That's why I removed it. But there may be cases where I missed so added it again https://github.com/wso2/carbon-apimgt/pull/10001/commits/c63181d28305d4b74edc9cc8bb186fe25117b703

```
            if (APIConstants.API_GLOBAL_VISIBILITY.equals(api.getVisibility())) { //global api
                return api;
            }
```

With this check, we are not achieving anything. Thus removed it.